include/bluetooth/mesh/gen_loc.h

```diff
@@ -14,23 +14,32 @@
 #ifndef BT_MESH_GEN_LOC_H__
 #define BT_MESH_GEN_LOC_H__
 
+#include <float.h>
 #include <zephyr/bluetooth/mesh.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @defgroup bt_mesh_loc_altitude_defines Defines for altitude.
- * Boundaries and special values for the altitude.
+ * @defgroup bt_mesh_loc_location_defines Defines for locations.
+ * Boundaries and special values for the latitude, longitude, altitude and local North/East.
  * @{
  */
-/** Highest altitude that can be represented */
-#define BT_MESH_LOC_ALTITUDE_MAX 32765
-/** Altitude is unknown or not configured. */
-#define BT_MESH_LOC_ALTITUDE_UNKNOWN 0x7fff
-/** Altitude is out of bounds. */
-#define BT_MESH_LOC_ALTITUDE_TOO_LARGE 0x7ffe
+/** The Global Longitude is unknown or not configured. */
+#define BT_MESH_LOC_GLOBAL_LONGITUDE_UNKNOWN DBL_MAX
```

Comment:
I think I prefer using `NAN` instead here (`math.h`), more semantic and less likely that users somehow end up treating this token value as an actual longitude? If we require `NAN` to be defined we won't support `CONFIG_MINIMAL_LIBC` though, not sure if that is a concern. The default is `PICOLIBC`.

Reply:
... and it would also require extra code in addition to the range checking you already have to correctly convert from NaN to the `0x8000000` value. Now that I think more about it the NAN solves a non-existent issue by creating a bigger mess and more code. I'll approve this like it is : )